typed-racket-lib/typed-racket/base-env/unit-prims.rkt

```diff
@@ -0,0 +1,767 @@
+               (define runtime-id
+                 (local-expand #'unit-id (syntax-local-context) (kernel-form-identifier-list)))
+               (tr:unit:compound:expr-property
+                #`(#%expression
+                   (begin
+                     (void)
+                     (void)
+                     (void)
+                     ;; all imports
+                     (void #,@(map (lambda (id) 
+                                     #`(quote-syntax #,id))
+                                   imports))
+                     ;; all exports
+                     (void #,@(map (lambda (id) 
+                                     #`(quote-syntax #,id))
+                                   exports))
+                     (void (quote-syntax #,runtime-id))))
+                'infer)))))
+
+;; NOTE/TODO:
+;; - it seems that the docs for compound-unit/infer
+;;   suggest that imports are filled in from the 
+;;   static information bound to the unit-ids
+;;   but simple tests don't seem to confirm this
+(define-syntax (compound-unit/infer stx)
+  (syntax-parse stx
+    [(_ 
+      imports:compound-infer-imports
+      exports:compound-infer-exports
+      links:compound-infer-links)
+     (ignore
+      (tr:unit:compound-property
+       (quasisyntax/loc stx
+         (#%expression
+          (begin
+            (void #,@#'imports.import-link-ids
+                  #,@(attribute links.bound-link-ids))
+            (void #,@#'imports.import-sig-ids
+                  #,@(attribute links.bound-sig-ids))
+            (void #,@#'imports.import-link-ids) 
+            exports.export-links-or-sigs
+            #,(attribute links.link-table)
+            (untyped-compound-unit/infer
+             imports
+             exports 
+             links))))
+       'infer))]))
+
+(define-syntax (define-compound-unit/infer stx)
+  (syntax-parse stx
+    [(_ unit-name:id
+        imports:compound-infer-imports
+        exports:compound-infer-exports
+        links:compound-infer-links)
+     (quasisyntax/loc stx
+       (process-define-compound-unit
+        (void #,@#'imports.import-link-ids
+              #,@(attribute links.bound-link-ids))
+        (void #,@#'imports.import-sig-ids
+              #,@(attribute links.bound-sig-ids))
+        (void #,@#'imports.import-link-ids) 
+        exports.export-links-or-sigs
+        #,(attribute links.link-table)
+        (untyped-define-compound-unit/infer unit-name
+                                            imports
+                                            exports
+                                            links)))]))
+
+;; Ignoring renames/prefix/etc for now
+(define-syntax (unit-from-context stx)
+  (syntax-parse stx
+    [(_ sig:id)
+     (ignore
+      (tr:unit:from-context-property
+       (quasisyntax/loc stx
+         (#%expression
+          (begin
+            (void (quote-syntax sig))
+            (untyped-unit-from-context sig))))
+       #t))]))
+
+
```

Comment:
Nitpick: extraneous whitespace at EOF. In general, delete extraneous whitespace.